src/scene/mesh.js

```diff
@@ -981,6 +997,85 @@ class Mesh extends RefCountedObject {
         };
         this.indexBuffer[RENDERSTYLE_WIREFRAME] = wireBuffer;
     }
+
+    /**
+     * Builds the array of triangles of the mesh
+     */
+    buildTriangleArray() {
+        const triangles = this.triangles;
+        const positions = [];
+        const indices = [];
+        const numPositions = this.getPositions(positions);
+        const numIndices = this.getIndices(indices);
+        const points = this.points;
+        const numTriangles = numIndices / 3;
+
+        if (points.length === 0) {
+            let j = 0;
+            for (let i = 0; i < numPositions; i++) {
+                const v = new Vec3(positions[j], positions[j + 1], positions[j + 2]);
+                points.push(v);
+                j += 3;
+            }
+        } else {
+            let j = 0;
+            for (let i = 0; i < numPositions; i++) {
+                const v = points[i];
+                v.set(positions[j], positions[j + 1], positions[j + 2]);
+                j += 3;
+            }
+        }
+
+
+        if (triangles.length === 0) {
+            let j = 0;
+            for (let i = 0; i < numTriangles; i++) {
+                const triangle = new Tri(points[indices[j]], points[indices[j + 1]], points[indices[j + 2]]);
+                triangles.push(triangle);
+                j += 3;
+            }
+        } else {
+            let j = 0;
+            for (let i = 0; i < numTriangles; i++) {
+                const triangle = triangles[i];
+                triangle.vertex0.copy(points[indices[j]]);
+                triangle.vertex1.copy(points[indices[j + 1]]);
+                triangle.vertex2.copy(points[indices[j + 2]]);
+                triangle.resetEdges();
+                j += 3;
+            }
+        }
+
+    }
+
+    /**
+     * @param {Ray} ray - The ray to interesect the mesh with
+     * @returns {number} The distance between the ray origin and closest point of intersection
+     */
+    rayCast(ray) {
+
+        // if the BVH doesn't exist, build it
+        // TODO: rebuilding bvh when dirtyCount exceeds limit
+        // Currently bvh building is not efficient enough for this to be feasible in real-time
+        if (!this.bvh || this.dirtyCount > Infinity) {
```

Comment:
what does `this.dirtyCount > Infinity` mean?

Reply:
Added a comment to explain this - dirtyCount is the number of times the bvh has been refitted since it was last built. Since this is likely to result in it becoming outdated for animated meshes, dirtyCount can be used to rebuild the bvh as appropriate when this occurs, but rebuilding is not currently fast enough for this process to be seamless.

Reply:
but what does the comparison mean logically? Isn't `count ? Infinity` always false?

Reply:
Yeah I've left it like that for now so that the line isn't triggered - should I remove this feature completely?

Reply:
I'd suggest we remove it.
And create a new issue on the engine repo with what can be done to improve the PR (and link this PR in the issue). Mention this one bit, but also anything else / links to documents, performance improvement and so on.

Reply:
Done